crates/typst/src/eval/str.rs

```diff
@@ -116,13 +116,22 @@ impl Str {
         match pattern {
             StrPattern::Str(pat) => self.0.ends_with(pat.as_str()),
             StrPattern::Regex(re) => {
-                // Regex expressions return non-overlapping matches, adding an
-                // ending anchor forces our regex to find the end
-                let anchored = Regex::new(&format!("({})$", re.as_str())).unwrap();
-                anchored
-                    .find_iter(self)
-                    .last()
-                    .map_or(false, |m| m.end() == self.0.len())
+                let mut start_byte = 0;
+                while let Some(mat) = re.find_at(self, start_byte) {
+                    if mat.end() == self.0.len() {
+                        return true;
+                    }
+
+                    // There might still be a match overlapping this one, so
+                    // restart at the next code point
+                    if let Some(c) = &self[start_byte..].chars().next() {
```

Comment:
should `start_byte` be `mat.start()` here? otherwise, it doesn't really make sense to me.

Reply:
Shit, thanks. Leftover from the previous commit.